### PR TITLE
Add planGraspsAndPick method to move_group_interface

### PR DESCRIPTION
**Pull request by Jntzko**
This utilizes the new GraspPlanning service and provides a way to make use of external grasp
planners without also adding default grasps.

The service definition is proposed in:
https://github.com/ros-planning/moveit_msgs/pull/32

### CODE REVIEW

**Comment by v4hn** [comment]
In the latest discussion in #246 @davetcoleman argued that the `pick` function should retain its current meaning (at least in indigo), whereas I consider it quite problematic and a safety-issue that the user currently has no way to access the `GraspPlanning.srv` functionality of the `move_group` node (it will call an external service when no grasps have been specified) without also invoking a default grasp in edge-conditions.
The proposed additional functions get rid of this restriction and provide an interface in `move_group_interface::MoveGroup(Interface)` by which such a service is automatically invoked and the function call fails if the service is not available.

Note that this also provides a simpler call structure than the current usage: The service is called by the client code to compute grasps before invoking the pick pipeline.

File: moveit_ros/planning_interface/move_group_interface/src/move_group.cpp [hunk 1]
```diff
@@ -590,6 +595,47 @@ class MoveGroup::MoveGroupImpl
...
 
+  MoveItErrorCode planGraspsAndPick(const std::string &object)
+  {
+    moveit::planning_interface::PlanningSceneInterface psi;
+
+    std::map<std::string, moveit_msgs::CollisionObject> objects = psi.getObjects(std::vector<std::string>(1,object));
+
+    if (objects.size() < 1)
+    {
+      ROS_ERROR_STREAM("Asked for grasps for the object '" << object << "', but the object could not be found");
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/move_group_interface/src/move_group.cpp]: ROS_ERROR_STREAM_NAMED

**Comment by davetcoleman** [comment]
As previously stated, you're welcome to make your proposed functionality changes in Kinetic, but I like this work around for indigo, thank you.

File: moveit_ros/planning_interface/move_group_interface/src/move_group.cpp (hunk 1 above)
Comment (v4hn) [inline moveit_ros/planning_interface/move_group_interface/src/move_group.cpp]: hehe, I proposed that too, but actually *none* of the log outputs in this class are `_NAMED` at the moment, and it seemed weird to introduce one occurrence here.

Instead I would propose to name all of them in a later request.

**Comment by v4hn** [comment]
Sorry dave, I should have added a bit more of my reasoning above.
By now, I pretty much agree with your argument to keep the default grasp(s) in the `pick` function.
It *is* a nice show-case.
The point I am worried about is that users cannot use the `database_grasp_planning` interface without also invoking the default grasps. So *if* a users utilizes the service interface, they might get unwanted motion plans executed on their hardware if their service is not available for some reason (this happened over here at least twice within a few weeks).
The new interface proposed here changes this.
So if we want to simplify the behavior of the current `pick` function later on (still makes sense in my opinion), I would actually propose to remove the `database_grasp_planning`-service-part from the function instead. :cake:

**Comment by davetcoleman** [comment]
> I would actually propose to remove the database_grasp_planning-service-part from the function instead.

But that would again change functionality in indigo that someone might be using. You are proposing this for kinetic I assume?

Also @Jntzko can you add more comments for the ``pick`` function explaining how it is different than this new interface?

**Comment by v4hn** [comment]
> But that would again change functionality in indigo that someone might be using. You are proposing this for kinetic I assume?

Yes, of course. That's what I meant by "later on". :)

**Comment by Jntzko** [comment]
@davetcoleman I changed the comments as you proposed.

**Comment by Jntzko** [comment]
I have added the _NAMED log outputs.
This should be ready to merged.

**Comment by v4hn** [comment]
@davetcoleman All feedback has been addressed and @Jntzko also added the _NAMED change to this request. I approve.
As the `_NAMED` change was added after your review, I'll leave it to you to merge it.

**Comment by davetcoleman** [comment]
@v4hn @Jntzko I started to pick this to the kinetic branch but the recent rename of the move_group_interface makes it pretty tricky - I think you'll have to manually re-implement these changes in the new location. Can you please do this since you are more familiar with both changes? I picked to Jade already via PR. Thanks.

**Pull request merged**